server/methods/createDirectMessage.js

```diff
@@ -108,7 +108,11 @@ Meteor.methods({
 DDPRateLimiter.addRule({
 	type: 'method',
 	name: 'createDirectMessage',
-	connectionId() {
-		return true;
+	userId(userId) {
+		const user = RocketChat.models.Users.findOneById(userId);
```

Comment:
Please use `RocketChat.authz.hasPermission(userId, permissionNode)` instead of finding a user by id and checking the role.